### PR TITLE
fix(sync): adding instant sync logic

### PR DESCRIPTION
**Pull request by bassrock**
# Goal

Enable instant sync based on our event stream from the event bridge.

# Implementation Notes

There is some extra logic in the old web repo around a user_tokens table, we have really stopped using that table years ago, so I did not port its logic here.

Once this is deployed, we can also remove all the push logic from web repo around this.

### CODE REVIEW

File: servers/push-server/src/gcm.ts [hunk 1]
```diff
@@ -8,22 +8,12 @@ const app = initializeApp({
...
-}
-
 export const sendNotificationToDevice = (
   notification: any,
-  payload: Payload,
   token: any,
 ): Promise<string> => {
   const message: Message = {
     token: token,
-    data: {
```
Comment (bassrock) [inline servers/push-server/src/gcm.ts:23]: Clients no longer use these.

File: servers/push-server/src/worker.ts [hunk 2]
```diff
@@ -23,40 +19,13 @@ const processMessage = async (fullMessage: Message): Promise<void> => {
     level: 'info',
   });
 
-  if (target === TARGET_APNS) {
```
Comment (bassrock) [inline servers/push-server/src/worker.ts:26]: No longer used

File: servers/push-server/src/worker.ts [hunk 3]
```diff
@@ -23,40 +19,13 @@ const processMessage = async (fullMessage: Message): Promise<void> => {
...
-      token,
-      false,
-    );
-  } else if (target === TARGET_APNS_SILENT) {
+  if (target === TARGET_APNS_SILENT) {
     console.log('APNS SILENT push', token);
-    await apns.sendNotificationToDevice(
-      contents,
-      {
-        g: message.guid,
```
Comment (bassrock) [inline servers/push-server/src/worker.ts:44]: Clients don't rely on the guid payload anymore. (verified in android and ios codebase)

Comment (bassrock) [reply servers/push-server/src/worker.ts:44]: It also simplifies the required code here.

File: lambdas/instant-sync-events/src/clients.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+ * Create a db client for reads from readitla_ril-* databases
+ */
+export async function readClient(): Promise<knex.Knex<any, any[]>> {
+  if (readDb) return readDb;
+
+  const secret = await fetchSecret(config.dbSecretName);
+
+  readDb = createConnection({
+    host: secret.read_host,
+    port: 3306,
```
Comment (kschelonka) [inline lambdas/instant-sync-events/src/clients.ts:18]: is this not in the secret?

File: lambdas/instant-sync-events/src/handlerFn.ts [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  'FAVORITE_ITEM',
+  'UNFAVORITE_ITEM',
+  'ARCHIVE_ITEM',
+  'UNARCHIVE_ITEM',
+  'ADD_TAGS',
+  'REPLACE_TAGS',
+  'CLEAR_TAGS',
+  'REMOVE_TAGS',
+  'RENAME_TAG',
+  'DELETE_TAG',
```
Comment (kschelonka) [inline lambdas/instant-sync-events/src/handlerFn.ts:23]: 'UPDATE_TITLE' - new event

File: lambdas/instant-sync-events/src/clients.ts (hunk 4 above)
Comment (bassrock) [reply lambdas/instant-sync-events/src/clients.ts:18]: no lol idk i feel a db port is alweays the same

**Review by kschelonka — APPROVED** [review]

**Pull request merged**